x-pack/plugin/core/src/main/plugin-metadata/plugin-security.policy

```diff
@@ -9,7 +9,24 @@ grant {
   permission java.util.PropertyPermission "*", "read,write";
 
   // needed for multiple server implementations used in tests
-  permission java.net.SocketPermission "*", "accept,connect";
+  permission java.net.SocketPermission "*", "accept,connect,resolve";
+
+  // needed for GSSAPI bind to LDAP
+  permission javax.security.auth.AuthPermission "modifyPrincipals";
+  permission javax.security.auth.AuthPermission "modifyPrivateCredentials";
+  permission javax.security.auth.PrivateCredentialPermission "javax.security.auth.kerberos.KerberosKey * \"*\"", "read";
+  permission javax.security.auth.PrivateCredentialPermission "javax.security.auth.kerberos.KeyTab * \"*\"", "read";
+  permission javax.security.auth.PrivateCredentialPermission "javax.security.auth.kerberos.KerberosTicket * \"*\"", "read";
+  permission javax.security.auth.AuthPermission "doAs";
+  permission javax.security.auth.kerberos.ServicePermission "*","initiate,accept";
+
+  permission java.util.PropertyPermission "javax.security.auth.useSubjectCredsOnly","write";
+  permission java.util.PropertyPermission "java.security.krb5.conf","write";
+  permission java.util.PropertyPermission "sun.security.krb5.debug","write";
+  permission java.util.PropertyPermission "java.security.debug","write";
+
+  permission javax.security.auth.AuthPermission "createLoginContext.GSSAPIBindRequest";
+  permission javax.security.auth.AuthPermission "setLoginConfiguration";
```

Comment:
can't these be moved to `x-pack/plugin/security/src/main/plugin-metadata/plugin-security.policy` ?

Reply:
I think these are a lot of obscure permissions we're granting for _any_ plugin (that inherits `x-pack-plugin-core`). I worry especially about the maintainability. In this case I think we should move the `unboundid` dependency over to the security plugin. I think should not have been a dependecy in `plugin-core` in the first place. The only imported class from that dependecy in `plugin-core` is an enum, `LdapSearchScope`, which I think we should emulate our own and use that in `plugin-core` and adapt the APIs.

What do you think?